src/ScheduleFunctions.cpp

```diff
@@ -448,6 +448,7 @@ Stmt build_produce(Function f) {
         // Iterate through all of the input args to the extern
         // function building a suitable argument list for the
         // extern function call.
+        vector<string> buffer_names;
```

Comment:
Related to my comment below, I think that this vector should actually store Stmt objects containing calls to the annotate runtime functions. This way, the Variable Exprs can be re-used and we don't need to remember buffer names for so long.